### PR TITLE
Updated Contributing Guide

### PR DESCRIPTION
**Pull request by vinijaiswal**
## Description
1. Updated the linux foundation delta chapter link.
2. Updated the Communication part to provide prescriptive guidelines for potential contributors.
3. Updated contributor metrics.

## How was this patch tested?
Not Applicable.

## Does this PR introduce _any_ user-facing changes?
Users will see more prescriptive guidance and updated resources to contribute.

Signed-off-by: Vini Jaiswal

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
 - We will use the GitHub issue to discuss the feature and come to agreement.
 - This is to prevent your time being wasted, as well as ours.
 - The GitHub review process for major features is also important so that organizations with commit access can come to agreement on design.
 - If it is appropriate to write a design document, the document must be hosted either in the GitHub tracking issue, or linked to from the issue and hosted in a world-readable location. Examples of design documents include [sample 1](https://docs.google.com/document/d/16S7xoAmXpSax7W1OWYYHo5nZ71t5NvrQ-F79pZF6yb8), [sample 2](https://docs.google.com/document/d/1MJhmW_H7doGWY2oty-I78vciziPzBy_nzuuB-Wv5XQ8), and [sample 3](https://docs.google.com/document/d/19CU4eJuBXOwW7FC58uSqyCbcLTsgvQ5P1zoPOPgUSpI).
 - Specifically, if the goal is to add a new extension, please read the extension policy.
-- Small patches and bug fixes don't need prior communication.
+- Small patches and bug fixes don't need prior communication. If you identified a bug and have ways to solve it, please create an [issue](https://github.com/delta-io/delta/issues).
```
Comment (scottsand-db) [inline CONTRIBUTING.md:18]: if they have ways to solve the issue, shouldn't they create a PR instead of a github issue?

**Review by scottsand-db — APPROVED** [review]
LGTM with 1 nit/comment

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,21 +1,23 @@
...
+Delta Lake is supported by a wide set of developers from over 50 organizations across multiple repositories.  Since 2019, more than 190 developers have contributed to Delta Lake!  The Delta Lake community is growing by leaps and bounds with more than 6000 members in the [Delta Users slack](https://dbricks.co/delta-users-slack)).
 
 For more information, please refer to the [founding technical charter](https://delta.io/pdfs/delta-charter.pdf).
 
 # Communication
-- Before starting work on a major feature, please reach out to us via [GitHub](https://github.com/delta-io/delta/issues), [Slack](https://dbricks.co/delta-users-slack), [email](https://groups.google.com/g/delta-users), etc. We will make sure no one else is already working on it and ask you to open a GitHub issue.
+- Before starting work on a major feature, please reach out to us via [GitHub](https://github.com/delta-io/delta/issues), [Slack](https://delta-users.slack.com/join/shared_invite/zt-165gcm2g7-0Sc57w7dX0FbfilR9EPwVQ#/shared-invite/email), [email](https://groups.google.com/g/delta-users), etc. We will make sure no one else is already working on it and ask you to open a GitHub issue.
```
Comment (zsxwing) [inline CONTRIBUTING.md:11]: do we have a long term solution for slack link? It will expire and in this case we want to update it quickly.

Comment (vinijaiswal) [reply CONTRIBUTING.md:11]: - In the last workshop, we gave [this link]( https://delta-users.slack.com/) and it didn't work for users who never joined slack.
- So we gave [this link](https://join.slack.com/t/delta-users/shared_invite/zt-165gcm2g7-0Sc57w7dX0FbfilR9EPwVQ) to the new users who wanted to join slack.

Happy to use what you think would make sense @zsxwing.

**Comment by vinijaiswal** [comment]
@zsxwing and @scottsand-db Anything else required from me to merge this PR? Thank you!

File: CONTRIBUTING.md (hunk 1 above)
Comment (tdas) [inline CONTRIBUTING.md:18]:
```suggestion
- Small patches and bug fixes don't need prior communication. If you have identified a bug and have ways to solve it, please create an [issue](https://github.com/delta-io/delta/issues).
```

File: CONTRIBUTING.md (hunk 2 above)
Comment (zsxwing) [reply CONTRIBUTING.md:11]: Just checked `https://dbricks.co/delta-users-slack`. Currently it points to `https://join.slack.com/t/delta-users/shared_invite/zt-165gcm2g7-0Sc57w7dX0FbfilR9EPwVQ`. I assumed @dennyglee updated the pointed link regularly to keep `https://dbricks.co/delta-users-slack` always valid without updating this page.

If we switch back to `https://delta-users.slack.com/join/shared_invite/zt-165gcm2g7-0Sc57w7dX0FbfilR9EPwVQ#/shared-invite/email`, we will have to update this page every time when this link is about to expire.

Comment (dennyglee) [reply CONTRIBUTING.md:11]: Note, please update the link to https://go.delta.io/delta-users - thanks!

Comment (tdas) [reply CONTRIBUTING.md:11]: Yes. this is the link the website uses as well. thank you.

Comment (zsxwing) [inline CONTRIBUTING.md:11]:
```suggestion
- Before starting work on a major feature, please reach out to us via [GitHub](https://github.com/delta-io/delta/issues), [Slack](https://go.delta.io/delta-users), [email](https://groups.google.com/g/delta-users), etc. We will make sure no one else is already working on it and ask you to open a GitHub issue.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,21 +1,23 @@
...
 - We will use the GitHub issue to discuss the feature and come to agreement.
 - This is to prevent your time being wasted, as well as ours.
 - The GitHub review process for major features is also important so that organizations with commit access can come to agreement on design.
 - If it is appropriate to write a design document, the document must be hosted either in the GitHub tracking issue, or linked to from the issue and hosted in a world-readable location. Examples of design documents include [sample 1](https://docs.google.com/document/d/16S7xoAmXpSax7W1OWYYHo5nZ71t5NvrQ-F79pZF6yb8), [sample 2](https://docs.google.com/document/d/1MJhmW_H7doGWY2oty-I78vciziPzBy_nzuuB-Wv5XQ8), and [sample 3](https://docs.google.com/document/d/19CU4eJuBXOwW7FC58uSqyCbcLTsgvQ5P1zoPOPgUSpI).
 - Specifically, if the goal is to add a new extension, please read the extension policy.
-- Small patches and bug fixes don't need prior communication.
+- Small patches and bug fixes don't need prior communication. If you have identified a bug and have ways to solve it, please create an [issue](https://github.com/delta-io/delta/issues).
```
Comment (zsxwing) [inline CONTRIBUTING.md:18]:
```suggestion
- Small patches and bug fixes don't need prior communication. If you have identified a bug and have ways to solve it, please create an [issue](https://github.com/delta-io/delta/issues) or create a [pull request](https://github.com/delta-io/delta/pulls).
```

**Comment by vinijaiswal** [comment]
All changes incorporated and resolved. Should I close the PR @zsxwing and @scottsand-db ? Thank you.

**Review by zsxwing — APPROVED** [review]
Thanks! I will merge this.

**Pull request closed without merging**